AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttV1Transport.cs

```diff
@@ -137,7 +140,55 @@ public override async Task SendIntrospection()
             string topic = _baseTopic + "/" + astarteInterface.InterfaceName + path;
             byte[] payload = AstartePayload.Serialize(value, timeStamp);
 
-            await DoSendMqttMessage(topic, payload, qos);
+            try
+            {
+                await DoSendMqttMessage(topic, payload, qos);
+            }
+            catch (MqttCommunicationException e)
+            {
+
+                HandleDatastreamFailedPublish(e, mapping, topic, payload, qos);
+            }
+        }
+
+        public override void RetryFailedMessages()
+        {
+            if (_failedMessageStorage is null)
+            {
+                return;
+            }
+
+            while (_failedMessageStorage.IsEmpty())
```

Comment:
I think should be add `!` here.
```suggestion
            while (!_failedMessageStorage.IsEmpty())
```